presto-main/src/test/java/com/facebook/presto/sql/planner/TestPlanDsl.java

```diff
@@ -0,0 +1,193 @@
+    {
+        assertPlan("SELECT extendedprice, orderkey, discount, orderkey, linenumber FROM lineitem",
+                output(ImmutableList.of("ORDERKEY", "ORDERKEY"),
+                        /*
+                         * This is a project node, but this gives us a convenient way to verify that
+                         * visitProject is correctly handled through an anyTree.
+                         */
+                        anyTree(
+                                tableScan("lineitem", ImmutableMap.of("ORDERKEY", "orderkey")))));
+    }
+
+    @Test
+    public void testUnreferencedSymbolsDontNeedBinding()
+    {
+        assertPlan("SELECT orderkey, 2 FROM lineitem",
+                output(ImmutableList.of("ORDERKEY"),
+                        anyTree(
+                                tableScan("lineitem", ImmutableMap.of("ORDERKEY", "orderkey")))));
+    }
+
+    @Test
+    public void testAliasConstantFromProject()
+    {
+        assertPlan("SELECT orderkey, 2 FROM lineitem",
+                output(ImmutableList.of("ORDERKEY", "TWO"),
+                        project(ImmutableMap.of("TWO", expression("2")),
+                                tableScan("lineitem", ImmutableMap.of("ORDERKEY", "orderkey")))));
+    }
+
+    @Test
+    public void testAliasExpressionFromProject()
+    {
+        assertPlan("SELECT 1 + orderkey FROM lineitem",
+                output(ImmutableList.of("EXPRESSION"),
+                        project(ImmutableMap.of("EXPRESSION", expression("1 + ORDERKEY")),
+                                tableScan("lineitem", ImmutableMap.of("ORDERKEY", "orderkey")))));
+    }
+
+    @Test
+    public void testTableScan()
+    {
+        assertPlan("SELECT orderkey FROM lineitem",
+                output(ImmutableList.of("ORDERKEY"),
+                        tableScan("lineitem", ImmutableMap.of("ORDERKEY", "orderkey"))));
+    }
+
+    @Test
+    public void testJoinMatcher()
+    {
+        assertPlan("SELECT o.orderkey FROM orders o, lineitem l WHERE l.orderkey = o.orderkey",
+                anyTree(
+                        join(INNER, ImmutableList.of(equiJoinClause("ORDERS_OK", "LINEITEM_OK")),
+                                any(
+                                        tableScan("orders").withAlias("ORDERS_OK", columnReference("orders", "orderkey"))),
+                                anyTree(
+                                        tableScan("lineitem").withAlias("LINEITEM_OK", columnReference("lineitem", "orderkey"))))));
+    }
+
+    @Test
+    public void testAggregation()
+    {
+        assertPlan("SELECT COUNT(nationkey) FROM nation",
+                output(ImmutableList.of("FINAL_COUNT"),
+                        aggregation(ImmutableMap.of("FINAL_COUNT", functionCall("count", ImmutableList.of("PARTIAL_COUNT"))),
+                                any(
+                                        aggregation(ImmutableMap.of("PARTIAL_COUNT", functionCall("count", ImmutableList.of("NATIONKEY"))),
```

Comment:
Yes. We'd have to magically track NATIONKEY through the first aggregation otherwise, and that's probably too much magic to be reliable.